lib/core/utils/parse-sameorigin-stylesheets.js

```diff
@@ -0,0 +1,94 @@
+/**
+ * Parse non cross-origin stylesheets
+ *
+ * @param {Object} sheet CSSStylesheet object
+ * @param {Object} options options object from `axe.utils.parseStylesheets`
+ * @param {Array<Number>} priority sheet priority
+ * @param {Array<String>} importedUrls urls of already imported stylesheets
+ * @param {Boolean} isCrossOrigin boolean denoting if a stylesheet is `cross-origin`
+ * @returns {Promise}
+ */
+axe.utils.parseSameOriginStylesheets = function parseSameOriginStylesheets(
+	sheet,
+	options,
+	priority,
+	importedUrls,
+	isCrossOrigin = false
+) {
+	const rules = Array.from(sheet.cssRules);
+
+	if (!rules) {
+		return Promise.resolve();
+	}
+
+	/**
+	 * reference -> https://developer.mozilla.org/en-US/docs/Web/API/CSSRule#Type_constants
+	 */
+	const cssImportRules = rules.filter(r => r.type === 3); // type === 3 -> CSSRule.IMPORT_RULE
+
+	/**
+	 * when no `@import` rules in given sheet -> resolve the current `sheet` & exit
+	 */
+	if (!cssImportRules.length) {
+		// exit
+		return Promise.resolve({
+			isCrossOrigin,
+			priority,
+			root: options.rootNode,
+			shadowId: options.shadowId,
+			sheet
+		});
+	}
+
+	/**
+	 * filter rules that are not already fetched
+	 */
+	const cssImportUrlsNotAlreadyImported = cssImportRules
+		// ensure rule has a href
+		.filter(rule => rule.href)
+		// extract href from object
+		.map(rule => rule.href)
+		// only href that are not already imported
+		.filter(url => !importedUrls.includes(url));
+
+	/**
+	 * iterate `@import` rules and fetch styles
+	 */
+	const promises = cssImportUrlsNotAlreadyImported.map(
+		(importUrl, cssRuleIndex) => {
+			const newPriority = [...priority, cssRuleIndex];
+			const isCrossOriginRequest = /^https?:\/\/|^\/\//i.test(importUrl);
```

Comment:
So even if the final url is the same origin as the current page, this check will mark it as cross origin.

So if the page is `http://example.com` and has a stylesheet url of `http://example.com/styles.css`, it will be treated as cross origin (even though it is same origin).

Reply:
Nope. 

The lines here - https://github.com/dequelabs/axe-core/pull/1339/files/bb47753e63ef18f52d17f3efd4336635062509f7#diff-c5302a50f699518e36fca8d4a242798dR46, does the filtering to ensure only `url's` that are not from the same origin flow into this code path.